test/spec/poc-common/clinical-services/clinicalServicesService.spec.js

```diff
@@ -209,4 +209,62 @@ describe('clinicalServicesService', function () {
 
   });
 
+  describe('getCsWithEncountersForPatient', function () {
+
+      var deferred;
+
+      var patient = {uuid: '5dc89638-b0cd-4390-bf10-ad07ed97965b'};
+      var encounterType = {uuid: "adultFolowup"}
+
+      var representation = 'custom:(uuid,encounterDatetime,obs:(value,concept:(display,uuid,mappings:(' +
+              'conceptReferenceTerm:(conceptSource:(display,uuid))))),provider:(display))';
+
+      var encounters = [{encounterDatetime: "today", obs: [{concept:{uuid: "poc-mapping-vitals"}}]}, 
+                        {encounterDatetime: "yesterday", obs: [{concept:{uuid: "poc-mapping-vitals"}}]}];
+      
+      var service = {markedOn: "poc-mapping-vitals"};
+
+      beforeEach(function () {
+        deferred = $q.defer();
+
+        spyOn(visitService, 'getTodaysVisit').and.callFake(function () {
+          return $q(function (resolve) {
+            return resolve({});
+          });
+        });
+        spyOn(encounterService, 'getEncountersForPatientByEncounterType').and.callFake(function () {
+          return $q(function (resolve) {
+            return resolve(encounters);
+          });
+        });
+      });
+
+      it('should get a response from getCsWithEncountersForPatient, with existing service', function () {
+
+          var returnedValue = null;
+
+          returnedValue = clinicalServicesService.getClinicalServiceWithEncountersForPatient(patient, service);
```

Comment:
Guys I don't think this test should be improved. ```returnedValue``` is actually a promise, so what this test and the other one is doing is checking if ```clinicalServicesService.getClinicalServiceWithEncountersForPatient``` returns a promise. The code is hard enough to understand and the test is not helping. If someone has a better understanding of the code please help making this test better.